### ISSUE TITLE
Feedback wanted: Would you use this?

### ISSUE THREAD

**Issue by armenzg** [getsentry/sentry-github-actions-app#46]
This project has been originally written for internal use in order to answer questions about getsentry's CI usage.

We would like to know if there's enough external interest to make this an official Github App that you can install on your org/repo.

Currently, you would need to deploy the app, configure your own Github App and then install it. If invested more on this project, we would get rid of the first two steps and all you would have to do is install the app.

If you would like more investment for this project, please vote this issue.

You are also welcome to set it up and give us some feedback. If you can deploy the app somewhere, the set up doesn't take very long.

**Comment by BYK** [comment]
I want this so badly so 👍🏻

**Comment by armenzg** [comment]
> I want this so badly so 👍🏻

Super! Let me know if the installations works for you.
There's also instructions on how to set up the app locally. Let me know if you ever try them.

**Comment by gunta** [comment]
Yes! It's something needed for improving DevOps four keys
https://github.com/dora-team/fourkeys

**Comment by armenzg** [comment]
> Yes! It's something needed for improving DevOps four keys https://github.com/dora-team/fourkeys

Let me know how well it works for you!

**Comment by JonasBa** [comment]
I was thinking about a similar app like this for the upcoming hackweek. Have we thought about parsing logs to find error stack traces and creating sentry errors for them?

**Comment by ChrisCarini** [comment]
> Would you use this?

Yes! Absolutely! ❤️ I just installed the app on several of my OSS repos. I'm an author and maintainer of a handful of plugins for JetBrains IDEs (IntelliJ, PyCharm, etc etc), as well as several other related, and those projects are the repos I'm particularly interested in using Sentry.io (and this GitHub app) with - finding this app is what caused me to sign up for Sentry! 🎉 

_(Note, I opened https://github.com/getsentry/sentry-github-actions-app/pull/67 for a small `README.md` doc improvement. In pursuit of starting out with just a few of my repos, I opted to select "Only Selected Repositories" - I initially did not give the app access to the `.sentry` repo I had created, and no telemetry was getting sent! 🤦 oops! Once I granted the app access to my private `.sentry` repo, everything worked as-expected! 🎉 )_

I have a handful of questions/suggestions/feedback - would you prefer I open separate issues for each, or would you rather discuss here? Just so I don't forget them, I'll list them here, but let me know if separate issues is preferred and I'll happily open them!

1. Is there a way to have the app map to different projects in Sentry? (custom `sentry_config.ini` on a per-GH-repo basis? a unique configuration in `.sentry` repository's `sentry_config.ini`? something else?)
2. I noticed the docs (`README.md`) say that this project's best value comes from "Discover" and "Dashboards" access, and then states that these two features are only available with "Business or Trial" and "Team" (respectively) - is that still the case? I believe I signed up for a "Developer" account, so once 14 days passes, I won't have access to this anymore? _(Note, I saw [Sentry gives free accounts for OSS projects](https://sentry.io/for/open-source/), and was considering applying for that - would the OSS account provide the ability for me to leverage the benefits of this app?)_
3. After installing the GitHub app, it landed me on the [GithubAppPostInstall.md](https://github.com/getsentry/sentry-github-actions-app/blob/main/GithubAppPostInstall.md) page, which states _"Currently the Github App is only meant to be used for internal use."_ - is that still the case? If not, I'm happy to submit a PR for this file to remove that, and add a few 'tips/suggestions' from myself from a 'newcomer' to Sentry/the GitHub app if you're open to it!

Best,
Chris Carini

**Comment by fbarbare** [comment]
That app is great! I would really love to see this being maintained and maybe expended to other CI tools ❤

**Comment by vipulgupta2048** [comment]
> I was thinking about a similar app like this for the upcoming hackweek. Have we thought about parsing logs to find error stack traces and creating sentry errors for them?

Very interested in Sentry being a CI/CD observability tool. Especially this. At first glance, it doesn't seem you can't do that with the current GitHub app not sending errors into Sentry and from the definition of issues in Sentry. It seems that Sentry creates issues automatically for errors that are present often. But, if everything gets sent on the `info` level, then Sentry is probably not recognizing any errors to create these issues. Would love to explore how we can send errors to Sentry.

**Comment by armenzg** [comment]
My apologies to everyone who commented on this ticket!! I had the wrong notification settings for this repo.

I have brought up this project back to the attention of my team and see if we can agree how it can fit with Sentry's priorities (this project was one of my hackweek projects for when I worked for the dev-infra team at Sentry).

**Comment by armenzg** [comment]
> I was thinking about a similar app like this for the upcoming hackweek. Have we thought about parsing logs to find error stack traces and creating sentry errors for them?

I have thought about it. I was part of the team that maintained the same idea for [Mozilla's CI](https://treeherder.mozilla.org/).

**Comment by cobyeastwood183** [comment]
https://sentry.zendesk.com/agent/tickets/147280

**Comment by ChrisCarini** [comment]
> My apologies to everyone who commented on this ticket!!

Don't sweat it! I'm personally just glad to get a response! :) Please keep us posted - it would be super cool to be able to leverage Sentry for GHA CI!

**Comment by philprime** [comment]
I really like what you all built here! 🙌 

Setup was straightforward and the activity is coming in perfect to explore.
I am using it for public and private projects at [@kula-app](https://github.com/kula-app) and [@techprimate](https://github.com/techprimate).

Feel free to reach out if you need additional use cases to explore.

**Comment by garbas** [comment]
I've finally setup this app, and now I want more from it :)
@armenzg this is great.

Can we also trace other events like issues/comments?

**Comment by armenzg** [comment]
Hi @garbas 👋🏻 
There's a new team around that _may_ have their goals line up with this.

With regards to your use case, what do you envision? I'm having trouble following it.

**Comment by douglasg14b** [comment]
I would, but not as a 3rd aprty app, we would need to be able to build and deploy this internally, and point to sentry.

**Comment by niebloomj** [comment]
I'm so excited I found this project, I really hope you invest in it especially with all the ai action we are now running this is key! I need seer to have this context!

**Comment by niebloomj** [comment]
Is there a way to create issues when jobs fail so I can have seer try to fix my failing workflow?

**Comment by armenzg** [comment]
> Is there a way to create issues when jobs fail so I can have seer try to fix my failing workflow?

There's isn't one. Could you please file a ticket to track it? It shouldn't be too hard.
It would require fetching the log upon a failure, iterate through each line & logging them and then capturing it as an error.

FYI I don't know if Sentry is currently investing in this project. I can help with reviews though.

**Comment by ckseet** [comment]
Is this ready for external use or still working for internal use
* I tried installing within an organization - and tried both single repo(action and .sentry) as well as all repos
* Did not receive any data on the sentry project
* Also noticed there are no webhooks configured after installation on either organization or repo level
Use case specifically interested in
* Is the time taken for actions steps (2nd picture under examples in the readme)

Or are there additional steps not documented in the readme?
